library/src/test/java/com/mindera/skeletoid/logs/appenders/LogFileAppenderUnitTest.java

```diff
@@ -0,0 +1,131 @@
+
+import org.junit.Test;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class LogFileAppenderUnitTest {
+
+    private String PACKAGE_NAME = "my.package.name";
+    private final String FILE_NAME = "FILE_NAME";
+
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorAllNull() {
+        LogFileAppender appender = new LogFileAppender(null, null);
+
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorPackageNameNull() {
+        LogFileAppender appender = new LogFileAppender(null, FILE_NAME);
+
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorFileNameInvalid() {
+        LogFileAppender appender = new LogFileAppender(PACKAGE_NAME, "*");
+
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructorFileNameNull() {
+        LogFileAppender appender = new LogFileAppender(PACKAGE_NAME, null);
+
+    }
+
+    @Test
+    public void testFileNameIsValidEmpty() {
+        LogFileAppender appender = new LogFileAppender(PACKAGE_NAME, FILE_NAME);
+        assertFalse(appender.isFilenameValid(""));
+    }
+
+    @Test
+    public void testFileNameIsValidInvalid() {
+        LogFileAppender appender = new LogFileAppender(PACKAGE_NAME, FILE_NAME);
+        assertFalse(appender.isFilenameValid("*"));
+    }
+
+    @Test
+    public void testFileNameIsValid() {
+        LogFileAppender appender = new LogFileAppender(PACKAGE_NAME, FILE_NAME);
+        assertTrue(appender.isFilenameValid(FILE_NAME));
+    }
+
+
+    @Test
+    public void testConstructor() {
+        LogFileAppender appender = new LogFileAppender(PACKAGE_NAME, FILE_NAME);
+
+        assertEquals("LogFileAppender", appender.getLoggerId());
+    }
+
+    @Test
+    public void testEnableAppender() {
+        Context context = mock(Context.class);
+
+        LogFileAppender appender = new LogFileAppender(PACKAGE_NAME, FILE_NAME);
+
+        appender.enableAppender(context);
+
+        //TODO What should I validate here?
+        verify(appender, times(1)).enableAppender(context);
+    }
+
+
+    @Test
+    public void testDisableAppender() {
+        Context context = mock(Context.class);
+
+        LogFileAppender appender = new LogFileAppender(PACKAGE_NAME, FILE_NAME);
+        appender.disableAppender();
+
+        //TODO What should I validate here?
+        verify(appender, times(1)).disableAppender();
+    }
+
+
+    //TODO How can I handle the date part on this?
+//    @Test
+//    public void testFormatLog() {
+//        LogFileAppender appender = new LogFileAppender(PACKAGE_NAME, FILE_NAME);
+//        assertEquals("", appender.formatLog(LOG.PRIORITY.DEBUG, null, "Hello", "My friend"));
```

Comment:
I would suggest to use a regex, something like `assertTrue(appender.formatLog.matches("xxx"));`
What do you think?